From 75c5094c4858f9ec9cc4a3bd376a8eaf32cdeaf3 Mon Sep 17 00:00:00 2001
From: Arturo Rinaldi <arty.net2@gmail.com>
Date: Thu, 21 Sep 2017 02:25:54 +0200
Subject: [PATCH 1/3] launcher.c: include <sys/ttydefaults.h>

This include is needed to get the definition of TTYDEF_*.

In addition to this, we also force UTMPX not to be defined as default
---
 shellinabox/launcher.c | 10 +++++++---
 1 file changed, 7 insertions(+), 3 deletions(-)

diff --git a/shellinabox/launcher.c b/shellinabox/launcher.c
index ba54b36..f8b25be 100644
--- a/shellinabox/launcher.c
+++ b/shellinabox/launcher.c
@@ -82,6 +82,10 @@
 #include <sys/uio.h>
 #endif
 
+#ifndef TTYDEF_IFLAG
+#include <sys/ttydefaults.h>
+#endif
+
 #ifdef HAVE_UTIL_H
 #include <util.h>
 #endif
@@ -90,9 +94,9 @@
 #include <utmp.h>
 #endif
 
-#ifdef HAVE_UTMPX_H
-#include <utmpx.h>
-#endif
+#undef HAVE_UTMPX_H
+
+#define HAVE_PTSNAME_R 1
 
 #if defined(HAVE_SECURITY_PAM_APPL_H)
 #include <security/pam_appl.h>
-- 
2.14.1

